### DISCUSSION TITLE
Handling Unsupported Languages

### DISCUSSION THREAD

**Discussion by JulienCertain** [amannn/next-intl#1052] [Q&A]
### Description

Hello everyone!

I'm currently in the process of coding a landing page with i18n and a static export. However, I've encountered a problem. While I've successfully implemented support for German and English languages on my page, attempting to switch to an unsupported language via URL triggers an error on my website.

My question is: How can I effectively capture any attempt to switch to an unsupported language? I've experimented with various approaches including template , error file, global-error file, not found, 404 file, and 500 file but unfortunately, none of these solutions seem to be working as intended. Any insights or suggestions would be greatly appreciated!

### Verifications

- [x] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

https://github.com/JulienCertain/next-intl-mvp-slug-error

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. install dependencies
3. run "npm run dev"
4.  open url "localhost:3000/fr"

### Expected behaviour

Server Error

Error: Page "/[locale]/page" is missing param "/fr" in "generateStaticParams()", which is required with "output: export" config.
This error happened while generating the page. Any console logs will be displayed in the terminal window.

**Comment by amannn** [comment]
If you use `output: 'export'`, you won't have a generated page for `/fr`. Depending on where/how you host your website, the user might receive a 404 error when requesting this pathname.

I'll move this to a discussion since this is a usage question.

**Reply by JulienCertain** [reply]
Thanks for the reply, I'm hosting my website on a AWS web server, would be possible for the server to catch this error?
